lightning/src/ln/channelmanager.rs

```diff
@@ -3668,6 +3674,29 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 
 					true
 				});
+
+				channel_state.claimable_htlcs.retain(|payment_hash, htlcs| {
+					if htlcs.into_iter().all(|htlc| {
+						if let OnionPayload::Invoice(ref final_hop_data) = htlc.onion_payload {
+							if htlc.value < final_hop_data.total_msat {
```

Comment:
I feel like we should handle all-parts-received MPP payments and non-MPP payments the same - ie instead of just checking if this is an MPP payment here, we should check if we've actually received all the parts we need (by summing up the HTLCs for each given payment hash) before we timeout-fail parts.

Reply:
Yeah I think I made an incorrect assumption that if we did have all the parts by this point they would have been claimed and removed from this set before we even get here.

Reply:
Wait, scratch my comment. I understand now. :facepalm: